internal/provider/meta.go

```diff
@@ -599,9 +591,50 @@ func createChildToken(d *schema.ResourceData, c *api.Client, namespace string) (
 	return childToken, nil
 }
 
+// GetResourceDataStr returns the value for a given ResourceData field
+// If the value is the zero value, then it checks the environment variable. If
+// the environment variable is empty, the default dv is returned
+func GetResourceDataStr(d *schema.ResourceData, field, env, dv string) string {
+	if s := d.Get(field).(string); s != "" {
+		return s
+	}
+
+	if env != "" {
+		if s := os.Getenv(env); s != "" {
+			return s
+		}
+	}
+
+	// return default
+	return dv
+}
+
+// GetResourceDataInt returns the value for a given ResourceData field
+// If the value is the zero value, then it checks the environment variable. If
+// the environment variable is empty, the default dv is returned
+func GetResourceDataInt(d *schema.ResourceData, field, env string, dv int) int {
+	if v := d.Get(field).(int); v != 0 {
+		return v
+	}
+	if env != "" {
+		if s := os.Getenv(env); s != "" {
+			ret, err := strconv.Atoi(s)
+			if err == nil {
+				return ret
+			}
+			// swallow the error and return the default because that is the
+			// behavior we had when using SDKv2's schema.EnvDefaultFunc
+		}
+	}
+	// return default
+	return dv
+}
+
 func GetToken(d *schema.ResourceData) (string, error) {
 	if token := d.Get("token").(string); token != "" {
 		return token, nil
+	} else if token = os.Getenv(api.EnvVaultToken); token != "" {
+		return token, nil
 	}
```

Comment:
Wondering if you could replace this if else with `GetResourceDataStr()`?

Reply:
yes, I think so :)